src/ol/render/canvas/replay.js

```diff
@@ -525,7 +525,10 @@ ol.render.canvas.Replay.prototype.replay_ = function(
         context.lineJoin = /** @type {string} */ (instruction[4]);
         context.miterLimit = /** @type {number} */ (instruction[5]);
         if (ol.has.CANVAS_LINE_DASH) {
-          context.setLineDash(/** @type {Array.<number>} */ (instruction[6]));
+          var lineDash = instruction[6].map(function(dash) {
```

Comment:
Since this operation is on the hot path, it would be better to reuse the modified lineDash, instead of creating a new array every time. Also, there is no need to modify the lineDash array when the pixel ratio is 1.